ci/pkl-pipelines/pay-deploy/infra-drift-detector.pkl

```diff
@@ -0,0 +1,153 @@
+          ["DEPLOYMENT_PATH"] = "\(env.split("-")[0])/\(env)/environment/bastion"
+        }
+      }
+    }
+    on_failure = shared_resources.paySlackNotificationForFail("#govuk-pay-starling", "Drift detected for $BUILD_JOB_NAME <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse build #$BUILD_NAME>")
+    on_success = shared_resources.paySlackNotificationForSuccess("#govuk-pay-activity", "Terraform state consistent for $BUILD_JOB_NAME <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse build #$BUILD_NAME>")
+  }
+}
+
+  for (env in List("deploy", "staging", "production")) {
+    new {
+      name = "\(env)-check-for-zendesk-unsubscribers"
+      plan {
+        new InParallelStep {
+          in_parallel = new Listing<Step>{
+            (getStep("every-15-minutes")) { trigger = true }
+            getStep("pay-ci")
+          }
+        }
+        new TaskStep {
+          task = "assume-role"
+          file = "pay-ci/ci/tasks/assume-role.yml"
+          params {
+            when (env == "production") {
+              ["AWS_ROLE_ARN"] = "arn:aws:iam::((pay_aws_prod_account_id)):role/concourse"
+            }
+            else {
+              ["AWS_ROLE_ARN"] = "arn:aws:iam::((pay_aws_\(env)_account_id)):role/concourse"
+            }
+            ["AWS_ROLE_SESSION_NAME"] = "zendesk-unsubscriber-assume-role"
+          }
+        }
+        (loadVar("role", "assume-role/assume-role.json")) { format = "json" }
+        new TaskStep {
+          task = "check-if-unsubscribed"
+          file = "pay-ci/ci/tasks/check-sns-topic-unsubscribers.yml"
+          params {
+            ["AWS_ACCESS_KEY_ID"] = "((.:role.AWS_ACCESS_KEY_ID))"
+            ["AWS_SECRET_ACCESS_KEY"] = "((.:role.AWS_SECRET_ACCESS_KEY))"
+            ["AWS_SESSION_TOKEN"] = "((.:role.AWS_SESSION_TOKEN))"
+            ["AWS_REGION"] = "eu-west-1"
+            ["AWS_DEFAULT_REGION"] = "eu-west-1"
+            when (env=="deploy") {
+              ["REGIONS"] = "eu-west-1 eu-central-1 us-east-1"
+            }
+            else {
+              ["REGIONS"] = "eu-west-1 us-east-1"
+            }
+          }
+        }
+      }
+    on_failure = shared_resources.paySlackNotificationForFail("#govuk-pay-starling", "An SNS topic subscriber is unsubscribed - $BUILD_JOB_NAME <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse build #$BUILD_NAME>")
+    on_success = shared_resources.paySlackNotificationForSuccess("#govuk-pay-activity", "SNS topic subscribers consistent - $BUILD_JOB_NAME <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse build #$BUILD_NAME>")
+    }
```

Comment:
The original pipeline has an `on_error` block as well as `on_failure` and `on_success`. You might implement this as a  function much like `paySlackNotificationForSuccess`. It would be fine as a `local` if no other pipelines require it; but if they do it would be better in the shared resources.

Reply:
I've not seen any other pipelines use the `on_error` step to alert to Slack. Any idea if we need to do it here/everywhere else?

Reply:
We do need it here, but not other places. on_error is when something has gone wrong in concourse, normally we don't care enough to need that since it's rare, but in this case it's forming part of our protective monitoring, so it's really important we know if these are failing